Mathlib/Algebra/Order/Kleene.lean

```diff
@@ -133,6 +133,14 @@ scoped[Computability] attribute [simp] add_eq_sup
 
 theorem add_idem (a : α) : a + a = a := by simp
 
+lemma natCast_eq_one {n : ℕ} (nezero : n ≠ 0) : (n : α) = 1 := by
+  induction n, Nat.one_le_iff_ne_zero.mpr nezero using Nat.le_induction with
+  | base => exact Nat.cast_one
+  | succ x _ hx => rw [Nat.cast_add, hx, Nat.cast_one, add_idem 1]
+
+lemma ofNat_eq_one {n : ℕ} [n.AtLeastTwo] : (no_index (OfNat.ofNat n : α)) = 1 :=
+  natCast_eq_one <| Nat.not_eq_zero_of_lt Nat.AtLeastTwo.prop
+
 theorem nsmul_eq_self : ∀ {n : ℕ} (_ : n ≠ 0) (a : α), n • a = a
```

Comment:
Can you prove `natCast_eq_one` in terms of this or the other way around?

Reply:
I did that originally in the first commit but it required importing `Mathlib.Data.Nat.Cast.Basic` (at least to do so the way that was obvious to me via `nsmul_eq_mul`) so I reverted it in https://github.com/leanprover-community/mathlib4/pull/20244/commits/e1a13efcd88b7b88eaf7e22b3900e3984151ef3f because the bot complained that added a large import.

Reply:
I see, fair enough